zipline/pipeline/data/dataset.py

```diff
@@ -104,14 +115,19 @@ def __get__(self, instance, owner):
         We don't bind to datasets at class creation time so that subclasses of
         DataSets produce different BoundColumns.
         """
+        if self.currency_aware:
+            currency_conversion = None
+        else:
+            currency_conversion = Disallowed
```

Comment:
The reason I went with using a sentinel here is because I thought it might be confusing/excessive to have both a `currency_conversion` parameter and a new `currency_aware` parameter, so I tried to combine the two.